packages/manager/apps/pci-databases-analytics/src/lib/availabilitiesHelper.ts

```diff
@@ -234,48 +239,53 @@ const mapFlavor = (
   return treeFlavor;
 };
 
+// handle plan code with or without .apac
+const getPlanCodeFromString = (planCode: string) => {
+  const hasApac = planCode.includes('.apac');
+  const normalizedPlanCode = hasApac
+    ? planCode.replace(`.${PRICING_APAC_SUFFIX}`, '')
+    : planCode;
+
+  return (consumption: string) =>
+    `${PRICING_PREFIX}.${normalizedPlanCode}.${consumption}${
+      hasApac ? `.${PRICING_APAC_SUFFIX}` : ''
+    }`;
+};
+
 const setPrices = (
   availability: database.Availability,
   catalog: order.publicOrder.Catalog,
   plan: Plan,
   flavor: Flavor,
 ) => {
-  const prefix = `databases.${availability.engine.toLowerCase()}-${
-    availability.plan
-  }-${availability.specifications.flavor}`;
-  [flavor.pricing.hourly] = catalog.addons.find(
-    (a) => a.planCode === `${prefix}.hour.consumption`,
-  ).pricings;
+  // Default pricing if not found in the catalog
+  const defaultPricing = { price: NaN, tax: 0 } as order.publicOrder.Pricing;
```

Comment:
I think NaN values should be avoided, you cannot detect them with a typeof or anything else

Reply:
I can not use either null or 0, as it would incorrectly display prices at 0€, so I changed it to undefined.

Reply:
NaN isn't undefined, they are 2 different values

Reply:
I forgot to push my changes 😅  Sorry about that, it is now "undefined" instead of "NaN"